tests/components/rfxtrx/conftest.py

```diff
@@ -1,12 +1,61 @@
 """Common test tools."""
-from unittest import mock
+from datetime import timedelta
 
 import pytest
 
+from homeassistant.components import rfxtrx
+from homeassistant.setup import async_setup_component
+from homeassistant.util.dt import utcnow
+
+from tests.async_mock import patch
+from tests.common import async_fire_time_changed
+
 
 @pytest.fixture(autouse=True, name="rfxtrx")
-async def rfxtrx(hass):
+async def rfxtrx_fixture(hass):
     """Fixture that cleans up threads from integration."""
 
-    with mock.patch("RFXtrx.Connect") as connect, mock.patch("RFXtrx.DummyTransport2"):
-        yield connect.return_value
+    with patch("RFXtrx.Connect") as connect, patch("RFXtrx.DummyTransport2"):
+        rfx = connect.return_value
+
+        async def _signal_event(packet_id):
+            event = rfxtrx.get_rfx_object(packet_id)
+            await hass.async_add_executor_job(
+                rfx.event_callback, event,
+            )
+
+            await hass.async_block_till_done()
+            await hass.async_block_till_done()
+            return event
+
+        rfx.signal = _signal_event
+
+        yield rfx
+
+
+@pytest.fixture(name="rfxtrx_automatic")
+async def rfxtrx_automatic_fixture(hass, rfxtrx):
+    """Fixture that starts up with automatic additions."""
+
+    assert await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)